### PR TITLE
Brandon createpresenter4

### PR DESCRIPTION
**Pull request by branflake2267**
Yesterdays work, working on the project config model sucking vars from the form. Also working on the form logic copy from eclipse. (In the middle of the form logic at the moment)

### CODE REVIEW

File: src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        this.revealInSlot = revealInSlot;
+    }
+
+    @Override
+    public String toString() {
+        String s = "{ PresenterConfigModel: ";
+        s += "name=" + name + " ";
+        s += "path=" + path + " ";
+        s += " }";
+        return s;
```
Comment (spg) [inline src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java]: maybe `String.format("{ PresenterConfigModel: name=%s path=%s }", name, path)` would be more legible?

Comment (branflake2267) [inline src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java]: Good thinking. :)

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java [hunk 2]
```diff
@@ -16,26 +21,80 @@
...
+  private JCheckBox btnIsCrawlable;
+  private JCheckBox btnCodesplit;
+  private JCheckBox btnAddUihandlers;
+  private JCheckBox btnAddOnbind;
+  private JCheckBox btnUseManualReveal;
+  private JCheckBox btnPrepareFromRequest;
+  private JCheckBox btnIsASingleton;
+  private JCheckBox btnIsASingleton2;
+  private JCheckBox btnOverrideDefaultPopup;
+  private JCheckBox btnAddOnhide;
```
Comment (spg) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: I like to keep widget typing out of those field names. For example, `JCheckBox btnAddOnhide` < `JCheckBox addOnhide`. The second naming option (`addOnhide`) is more future-proof IMO. You won't have to rename it if ever you decide to move from a checkbox to a listbox or any other swing component.

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java [hunk 3]
```diff
@@ -16,26 +21,80 @@
...
 
   public CreatePresenterForm(Project project) {
     super(project);
     init();
 
     setTitle("Create GWTP Presenter");
+
+    presenterConfigModel = new PresenterConfigModel();
+
+    //initHandlers();
```
Comment (spg) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: commented code

Comment (branflake2267) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: Good point, I was in the middle of debugging. :)

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java (hunk 2 above)
Comment (branflake2267) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: I was waiting for this to come up sooner or latter. I'm copying auto generated vars from eclipse, and will rename them as soon as I'm done copying :)

Comment (branflake2267) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: Good job in catching. I was hoping to save some time with so I could figure out the wiring. But you're right, what you said is better.

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java [hunk 4]
```diff
@@ -16,26 +21,80 @@
...
+      }
+    });
+
+    tabbedPanel.addChangeListener(new ChangeListener() {
+      @Override
+      public void stateChanged(ChangeEvent e) {
+        int index = tabbedPanel.getSelectedIndex();
+        setPresenterType(index);
+        if (index == 0) {
+          btnNestedPresenter.setSelected(true);
```
Comment (spg) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: maybe store those 3 widgets in an array, and do something like

```
for(SomeCommonAncestorClass component : theComponentArray) {
   component.setSelected(false);
}

theComponentArray[index].setSelected(true);
```

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java [hunk 5]
```diff
@@ -43,4 +102,45 @@ public CreatePresenterForm(Project project) {
...
     return contentPane;
   }
+
+  private void createUIComponents() {
+    // TODO: place custom component creation code here
+  }
+
+  private void setPresenterType(int index) {
+    if (index == 0) {
+      presenterConfigModel.setNestedPresenter(true);
```
Comment (spg) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: same suggestion as above applies here

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java (hunk 4 above)
Comment (branflake2267) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: Good point. There probably is a Toggle Group that wires it up too, haven't looked yet for IDEA.

File: src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    public void setPlace(boolean place) {
+        this.place = place;
+    }
+
+    public String getNameToken() {
+        return nameToken;
+    }
+
+    public String getNameTokenMethodName() {
+        return "get" + nameToken.substring(0, 1).toUpperCase() + nameToken.substring(1);
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java:125]: can nameToken be empty or do you have validations happening beforehand?

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java [hunk 7]
```diff
@@ -43,4 +102,45 @@ public CreatePresenterForm(Project project) {
   protected JComponent createCenterPanel() {
     return contentPane;
   }
+
+  private void createUIComponents() {
+    // TODO: place custom component creation code here
+  }
+
+  private void setPresenterType(int index) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterForm.java]: this method could be extracted in a few private methods, it's too long IMO

**Pull request merged**